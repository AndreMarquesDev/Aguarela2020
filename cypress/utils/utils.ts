import { Locale } from '../../types/Locale';
import {
    menuIconButtonDataTestId,
    nukaCarouselNextButtonDataTestId,
} from '../../utils/dataTestIds';
import { TextsInterface, textsPt, textsEn } from '../../utils/texts';

export const getLocalizedTexts = (locale: Locale): TextsInterface => {
    if (locale === Locale.Pt) {
        return textsPt;
    }

    return textsEn;
};

export const openMenuMobile = (isMobile: boolean, isFirefox: boolean): void => {
    if (isMobile) {
        cy.getByDataTestId(menuIconButtonDataTestId).click({ force: isFirefox });
    }
};

export const clickNextArrowButtonIfMobile = (
    isMobile: boolean,
    container: string,
    containerTitle: string
): void => {
    const isFirefox = Cypress.isBrowser('firefox');

    if (isMobile) {
        cy.getByDataTestIdParent(container, nukaCarouselNextButtonDataTestId).click({
            force: isFirefox,
        });

        cy.getByText(container, containerTitle).scrollIntoView();
    }
};
