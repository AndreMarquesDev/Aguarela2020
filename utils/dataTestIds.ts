export const projectsListNoCarouselDataTestId = 'projectsList_noCarousel';
export const projectItemTouchDivDataTestId = 'projectItem_touch_div';
export const contactFormContainerDataTestId = 'contactForm_container';
export const servicesBlockItemWrapperDataTestId = 'servicesBlock_blockWrapper';
export const servicesBlockItemCarouselDataTestId = 'servicesBlock_blockCarousel';
export const skillsBlockItemWrapperDataTestId = 'skillsBlock_skillItemWrapper';
export const skillsBlockItemCarouselDataTestId = 'skillsBlock_skillItemCarousel';
export const textBlock1DataTestId = 'textShadowBlock_textBlock1';
