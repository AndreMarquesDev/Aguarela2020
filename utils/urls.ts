export const contactFormBackendUrl = '/api/submitContactForm';

export const aguarelaDigitalInstagramUrl = 'https://www.instagram.com/aguareladigital';
export const aguarelaDigitalFacebookUrl = 'https://www.facebook.com/aguareladigitalagency';
export const aguarelaDigitalEmail = 'catarina@aguareladigital.com';
export const catarinaSantiagoInstagramUrl = 'https://www.instagram.com/catarinasantiago';
export const andreMarquesDevWebsiteUrl = 'https://www.andremarquesdev.com';

export const avocadoInstagramUrl = 'https://www.instagram.com/avocadohouselisbon';
export const tjelaInstagramUrl = 'https://www.instagram.com/tudonatjela';
export const guacamoleInstagramUrl = 'https://www.instagram.com/guacamolegmg';
export const kaffeehausInstagramUrl = 'https://www.instagram.com/kaffeehaus_lisboa';
export const tasteOfIndiaInstagramUrl = 'https://www.instagram.com/tasteofindiapt';
export const icecreamRollInstagramUrl = 'https://www.instagram.com/icecreamroll.pt';
export const mariaLimaoInstagramUrl = 'https://www.instagram.com/bebemarialimao';
export const biergartenInstagramUrl = 'https://www.instagram.com/biergarten.jardimdacerveja';
export const trattoriaInstagramUrl = 'https://www.instagram.com/trattoriadipaola';
export const aAmigaEsteticistaInstagramUrl = 'https://www.instagram.com/aamigaesteticista';
export const riceMeDeliInstagramUrl = 'https://www.instagram.com/ricemedeli';
export const quatroPatasDe5EstrelasInstagramUrl = 'https://www.instagram.com/4patasde5estrelas';
export const riceMeInstagramUrl = 'https://www.instagram.com/ricemerestaurante';
export const luminousInstagramUrl = 'https://www.instagram.com/becomeluminouscoaching';
export const madMaryInstagramUrl = 'https://www.instagram.com/madmarycuisine';
export const bovineInstagramUrl = 'https://www.instagram.com/bovinelisboa';
export const jamesonInstagramUrl = 'https://www.instagram.com/jamesonportugal';

export const harpoonLinkedInUrl = 'https://www.linkedin.com/company/harpoon-lda';

export const pernodWebsiteUrl = 'https://www.pernod-ricard.com';
export const beefeaterWebsiteUrl = 'https://www.beefeater.pt';
export const mariaLimaoIndustriaCriativaUrl =
    'https://industriacriativa.pt/projeto/45759/maria-limao-gestao-de-redes-sociais';
export const aAmigaEsteticistaIndustriaCriativaUrl =
    'https://industriacriativa.pt/projeto/14473/a-amiga-esteticista';
export const anaroIndustriaCriativaUrl =
    'https://industriacriativa.pt/projeto/45758/anaro-artist-content-creation';
