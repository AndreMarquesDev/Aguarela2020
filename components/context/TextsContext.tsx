import { createContext, Context } from 'react';
import { TextsInterface, textsPt } from '../../utils/texts';

export interface TextsContextProps {
    texts: TextsInterface;
}

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
