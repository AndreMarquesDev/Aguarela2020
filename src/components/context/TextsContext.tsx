import type { Context } from 'react';
import type { TextsInterface } from '../../utils/texts';
import { createContext } from 'react';
import { textsPt } from '../../utils/texts';

export type TextsContextProps = {
    texts: TextsInterface;
};

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
