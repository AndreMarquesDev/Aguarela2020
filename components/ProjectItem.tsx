import React, { FC, useContext } from 'react';
import Image from 'next/image';
import TextsContext from './context/TextsContext';

interface ProjectItemProps {
    imageSrc: string;
    imageAlt: string;
    brandLink: string;
    brandTag: string;
    year: number;
    isDesktop: boolean;
    isInPartnership?: boolean;
}

const ProjectItem: FC<ProjectItemProps> = ({ imageSrc, imageAlt, brandLink, brandTag, year, isDesktop, isInPartnership }) => {
    const { texts } = useContext(TextsContext);

    return (
        <>
            {isDesktop ? (
                <li>
                    <Image
                        alt={imageAlt}
                        height={400}
                        src={imageSrc}
                        width={400}
                    />
                    <div className="backface">
                        <a className="brand link" href={brandLink} rel="noreferrer" target="_blank">{brandTag}</a>
                        <p className="description">{texts.socialMediaManagementAndContentCreation}</p>
                        <p className="date">
                            {texts.since}
                            {' '}
                            {year}
                        </p>
                        {isInPartnership && (
                            <small>
                                *
                                {' '}
                                {texts.inPartnershipWith}
                            </small>
                        )}
                    </div>
                </li>
            ) : (
                <div className="carouselItem">
                    <Image
                        alt={imageAlt}
                        height={400}
                        src={imageSrc}
                        width={400}
                    />
                    <div className="backface">
                        <a className="brand link" href={brandLink} rel="noreferrer" target="_blank">{brandTag}</a>
                        <p className="description">{texts.socialMediaManagementAndContentCreation}</p>
                        <p className="date">
                            {texts.since}
                            {' '}
                            {year}
                        </p>
                        {isInPartnership && (
                            <small>
                                *
                                {' '}
                                {texts.inPartnershipWith}
                            </small>
                        )}
                    </div>
                </div>
            )}

            <style jsx>
                {`
                    @import './styles/_vars.scss';

                    li {
                        max-height: 400rem;
                        position: relative;
                        margin-right: 2%;

                        &:last-child {
                            margin-right: 0;
                        }

                        &:hover {
                            .backface {
                                display: flex;
                            }
                        }
                    }

                    .carouselItem {
                        display: flex;
                        flex-direction: column;
                        align-items: center;
                        text-align: center;

                        &:hover {
                            .backface {
                                display: flex;
                            }
                        }
                    }

                    .backface {
                        display: none;
                        max-width: 400rem;
                        max-height: 400rem;
                        width: 100%;
                        height: 100%;
                        position: absolute;
                        top: 0;
                        left: 50%;
                        transform: translateX(-50%);
                        flex-direction: column;
                        justify-content: center;
                        align-items: center;
                        text-align: center;
                        background: $blue;
                        padding: 10rem;

                        .brand, .date {
                            @include fontM($yellow, uppercase);
                            margin-bottom: 20rem;

                            @include desktop {
                                @include fontS($yellow, uppercase);
                            }

                            @include phablet {
                                @include fontXS($yellow, uppercase);
                            }
                        }

                        .description {
                            @include fontM($yellow);
                            margin-bottom: 20rem;

                            @include desktop {
                                @include fontS($yellow);
                            }

                            @include phablet {
                                @include fontXS($yellow);
                            }
                        }

                        small {
                            @include fontXS($yellow);
                        }
                    }
                `}
            </style>
        </>
    );
};

export default ProjectItem;
