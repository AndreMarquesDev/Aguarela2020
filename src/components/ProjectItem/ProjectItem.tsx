import React, { useContext, TouchEvent } from 'react';
import Image from 'next/image';
import { isClientSide } from 'multilingual-url/lib';
import classNames from 'classnames';
import { TextsContext } from '../context/TextsContext';
import { resetTimesTouchedAttribute } from '../../utils/generic';
import { projectItemTouchDivDataTestId } from '../../utils/dataTestIds';
import { desktopMediaQuery, phabletMediaQuery, tabletMediaQuery } from '../../styles/mediaQueries';

export interface ProjectItemProps {
    imageSrc: string;
    imageAlt: string;
    brandLink: string;
    brandTag: string;
    description: string;
    isDesktop?: boolean;
    year?: number | string;
    isInPartnership?: boolean;
    isActive?: boolean;
    isGrid?: boolean;
    preloadImage?: boolean;
}

export const ProjectItem = ({
    imageSrc,
    imageAlt,
    brandLink,
    brandTag,
    description,
    year,
    isDesktop,
    isInPartnership,
    isActive,
    isGrid,
    preloadImage,
}: ProjectItemProps): JSX.Element => {
    const { texts } = useContext(TextsContext);

    const imageSize = 400;
    const isTouch = isClientSide && window?.matchMedia('(hover: none), (pointer: coarse)').matches;

    const handleTouch = (event: TouchEvent<HTMLElement>): void => {
        const target = event.currentTarget;
        const timesTouched = parseInt(target.dataset.timesTouched as string);

        if (timesTouched === 0) {
            resetTimesTouchedAttribute();
            target.dataset.timesTouched = '1';
        } else {
            (target.querySelector('a.brand.link') as HTMLAnchorElement).click();
        }
    };

    return (
        <>
            {isDesktop ? (
                <li>
                    <Image
                        alt={imageAlt}
                        height={imageSize}
                        priority={!!preloadImage}
                        src={imageSrc}
                        style={{
                            width: '100%',
                            height: 'auto',
                            position: 'relative',
                            maxWidth: `${imageSize}px`,
                        }}
                        width={imageSize}
                    />
                    <div className="backface">
                        <a
                            className="brand link fontXS"
                            href={brandLink}
                            rel="noreferrer"
                            target="_blank"
                        >
                            {brandTag}
                        </a>
                        <p className="description fontS">{description}</p>
                        {year && (
                            <p className="date fontXS">
                                {year}
                                {isActive && ` - ${texts.present}`}
                            </p>
                        )}
                        {isInPartnership && (
                            <small className="fontXS">* {texts.inPartnershipWith}</small>
                        )}
                    </div>
                </li>
            ) : (
                <div
                    className={classNames('carouselItem', isTouch && 'touch', isGrid && 'grid')}
                    data-testid={`${projectItemTouchDivDataTestId}_${imageAlt}`}
                    data-times-touched={0}
                    onTouchStart={handleTouch}
                >
                    <Image
                        alt={imageAlt}
                        height={imageSize}
                        priority={!!preloadImage}
                        src={imageSrc}
                        style={{
                            width: '100%',
                            height: 'auto',
                            position: 'relative',
                            maxWidth: `${imageSize}px`,
                        }}
                        width={imageSize}
                    />
                    <div className="backface">
                        <a
                            className="brand link fontXS"
                            href={brandLink}
                            rel="noreferrer"
                            target="_blank"
                        >
                            {brandTag}
                        </a>
                        <p className="description fontS">{description}</p>
                        {year && (
                            <p className="date fontXS">
                                {year}
                                {isActive && ` - ${texts.present}`}
                            </p>
                        )}
                        {isInPartnership && (
                            <small className="fontXS">* {texts.inPartnershipWith}</small>
                        )}
                    </div>
                </div>
            )}

            <style jsx>
                {`
                    li {
                        max-height: 400rem;
                        position: relative;
                        margin-right: 2%;

                        &:last-child {
                            margin-right: 0;
                        }

                        &:hover {
                            .backface {
                                display: flex;
                            }
                        }
                    }

                    .carouselItem {
                        max-width: 400rem;
                        max-height: 400rem;
                        width: auto;
                        height: auto;
                        display: flex;
                        flex-direction: column;
                        align-items: center;
                        text-align: center;
                        position: relative;
                        margin: 0 auto;

                        &:hover {
                            .backface {
                                display: flex;
                            }
                        }

                        &.grid {
                            margin: 0 4% 0;

                            @media (${tabletMediaQuery}) {
                                margin: 0 2% 0;
                            }

                            .backface {
                                .brand {
                                    color: var(--yellow);
                                    text-transform: uppercase;

                                    @media (${tabletMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }

                                .date {
                                    color: var(--yellow);

                                    @media (${tabletMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }

                                .description {
                                    color: var(--yellow);

                                    @media (${desktopMediaQuery}) {
                                        font-size: var(--fontSizeXS);
                                    }

                                    @media (${tabletMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }

                                small {
                                    color: var(--yellow);

                                    @media (${tabletMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }
                            }
                        }
                    }

                    .backface {
                        display: none;
                        width: 100%;
                        height: 100%;
                        position: absolute;
                        top: 0;
                        left: 0;
                        flex-direction: column;
                        justify-content: center;
                        align-items: center;
                        text-align: center;
                        background: var(--blue);
                        padding: 10rem;

                        .brand {
                            color: var(--yellow);
                            text-transform: uppercase;
                            margin-bottom: 20rem;
                        }

                        .date {
                            color: var(--yellow);
                            margin-bottom: 20rem;
                        }

                        .description {
                            color: var(--yellow);
                            margin-bottom: 20rem;

                            @media (${phabletMediaQuery}) {
                                font-size: var(--fontSizeXS);
                            }
                        }

                        small {
                            color: var(--yellow);
                        }
                    }
                `}
            </style>
        </>
    );
};
