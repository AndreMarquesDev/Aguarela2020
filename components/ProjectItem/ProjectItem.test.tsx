import '@testing-library/jest-dom';
import React from 'react';
import { render, screen, RenderResult, fireEvent } from '@testing-library/react';
import { ProjectItem, ProjectItemProps } from './ProjectItem';
import { textsEn, textsPt } from '../../utils/texts';
import TextsContext from '../context/TextsContext';
import { projectItemTouchDivDataTestId } from '../../utils/dataTestIds';
import { tjelaInstagramUrl } from '../../utils/urls';

const baseProps: ProjectItemProps = {
    brandLink: tjelaInstagramUrl,
    brandTag: '@tudonatjela',
    description: textsPt.socialMediaManagementAndContentCreation,
    imageAlt: 'tjela logo',
    imageSrc: '/images/tjela.jpg',
    isDesktop: true,
    isInPartnership: true,
    isActive: true,
    isGrid: true,
    year: 2020,
};

const renderComponent = (newProps?: Partial<ProjectItemProps>): RenderResult => {
    return render(<ProjectItem {...baseProps} {...newProps} />);
};

describe('<ProjectItem />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        const image = screen.getByAltText(baseProps.imageAlt);
        const brandLink = screen.getByText(baseProps.brandTag);
        const description = screen.getByText(baseProps.description);
        const year = screen.getByText(`${baseProps.year} - ${textsPt.present}`);
        const isInPartership = screen.getByText(`* ${textsPt.inPartnershipWith}`);

        expect(image).toBeInTheDocument();
        expect(brandLink).toBeInTheDocument();
        expect(description).toBeInTheDocument();
        expect(year).toBeInTheDocument();
        expect(isInPartership).toBeInTheDocument();

        expect(brandLink).toHaveAttribute('href', baseProps.brandLink);

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        const { container } = render(
            <TextsContext.Provider
                value={{
                    texts: textsEn,
                }}
            >
                <ProjectItem
                    {...baseProps}
                    description={textsEn.socialMediaManagementAndContentCreation}
                />
            </TextsContext.Provider>
        );

        const year = screen.getByText(`${baseProps.year} - ${textsEn.present}`);
        const isInPartership = screen.getByText(`* ${textsEn.inPartnershipWith}`);

        expect(year).toBeInTheDocument();
        expect(isInPartership).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in mobile', () => {
        const { container } = renderComponent({ isDesktop: false });

        const image = screen.getByAltText(baseProps.imageAlt);
        const brandLink = screen.getByText(baseProps.brandTag);
        const description = screen.getByText(baseProps.description);
        const year = screen.getByText(`${baseProps.year} - ${textsPt.present}`);
        const isInPartership = screen.getByText(`* ${textsPt.inPartnershipWith}`);

        expect(image).toBeInTheDocument();
        expect(brandLink).toBeInTheDocument();
        expect(description).toBeInTheDocument();
        expect(year).toBeInTheDocument();
        expect(isInPartership).toBeInTheDocument();

        expect(brandLink).toHaveAttribute('href', baseProps.brandLink);

        expect(container).toMatchSnapshot();
    });

    test('renders properly with the year as a string', () => {
        renderComponent({ year: '2019 - 2020' });

        const year = screen.getByText(`2019 - 2020 - ${textsPt.present}`);

        expect(year).toBeInTheDocument();
    });

    test('renders properly with isActive being false', () => {
        renderComponent({ isActive: false });

        const year = screen.getByText(baseProps.year);

        const presentText = screen.queryByText(textsPt.present);

        expect(year).toBeInTheDocument();

        expect(presentText).not.toBeInTheDocument();
    });

    test('renders properly with isInPartnership being false', () => {
        renderComponent({ isInPartnership: false });

        const isInPartership = screen.queryByText(`* ${textsPt.inPartnershipWith}`);

        expect(isInPartership).not.toBeInTheDocument();
    });

    test('renders properly with isGrid being false', () => {
        const { container } = renderComponent({ isGrid: false, isDesktop: false });

        expect(container).toMatchSnapshot();
    });

    test('renders properly when device is touch', () => {
        window.matchMedia = jest.fn().mockImplementation(() => ({
            matches: true,
        }));

        const { container } = renderComponent({ isDesktop: false });

        const wrapper = screen.getByTestId(projectItemTouchDivDataTestId);

        expect(wrapper).toHaveAttribute('data-times-touched', '0');

        fireEvent.touchStart(wrapper);

        expect(wrapper).toHaveAttribute('data-times-touched', '1');

        fireEvent.touchStart(wrapper);

        expect(wrapper).toHaveAttribute('data-times-touched', '1');

        expect(container).toMatchSnapshot();
    });
});
