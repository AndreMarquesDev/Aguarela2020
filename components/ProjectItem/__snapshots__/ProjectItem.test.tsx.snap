// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectItem /> renders properly 1`] = `
<div>
  <li
    class="jsx-4127294762"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAwIiBoZWlnaHQ9IjQwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="tjela logo"
        decoding="async"
        src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75"
        srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 2x"
        style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <div
      class="jsx-4127294762 backface"
    >
      <a
        class="jsx-4127294762 brand link"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-4127294762 description"
      >
        Gestão de Redes Sociais e Criação de Conteúdo
      </p>
      <p
        class="jsx-4127294762 date"
      >
        2020
         - Presente
      </p>
      <small
        class="jsx-4127294762"
      >
        * 
        em dupla com Sofia Ferreirinho
      </small>
    </div>
  </li>
</div>
`;

exports[`<ProjectItem /> renders properly in English 1`] = `
<div>
  <li
    class="jsx-4127294762"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAwIiBoZWlnaHQ9IjQwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="tjela logo"
        decoding="async"
        src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75"
        srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 2x"
        style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <div
      class="jsx-4127294762 backface"
    >
      <a
        class="jsx-4127294762 brand link"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-4127294762 description"
      >
        Social Media Management and Content Creation
      </p>
      <p
        class="jsx-4127294762 date"
      >
        2020
         - Present
      </p>
      <small
        class="jsx-4127294762"
      >
        * 
        in partnership with Sofia Ferreirinho
      </small>
    </div>
  </li>
</div>
`;

exports[`<ProjectItem /> renders properly in mobile 1`] = `
<div>
  <div
    class="jsx-4127294762 carouselItem grid"
    data-testid="projectItem_touch_div_tjela logo"
    data-times-touched="0"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAwIiBoZWlnaHQ9IjQwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="tjela logo"
        decoding="async"
        src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75"
        srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 2x"
        style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <div
      class="jsx-4127294762 backface"
    >
      <a
        class="jsx-4127294762 brand link"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-4127294762 description"
      >
        Gestão de Redes Sociais e Criação de Conteúdo
      </p>
      <p
        class="jsx-4127294762 date"
      >
        2020
         - Presente
      </p>
      <small
        class="jsx-4127294762"
      >
        * 
        em dupla com Sofia Ferreirinho
      </small>
    </div>
  </div>
</div>
`;

exports[`<ProjectItem /> renders properly when device is touch 1`] = `
<div>
  <div
    class="jsx-4127294762 carouselItem touch grid"
    data-testid="projectItem_touch_div_tjela logo"
    data-times-touched="1"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAwIiBoZWlnaHQ9IjQwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="tjela logo"
        decoding="async"
        src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75"
        srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 2x"
        style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <div
      class="jsx-4127294762 backface"
    >
      <a
        class="jsx-4127294762 brand link"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-4127294762 description"
      >
        Gestão de Redes Sociais e Criação de Conteúdo
      </p>
      <p
        class="jsx-4127294762 date"
      >
        2020
         - Presente
      </p>
      <small
        class="jsx-4127294762"
      >
        * 
        em dupla com Sofia Ferreirinho
      </small>
    </div>
  </div>
</div>
`;

exports[`<ProjectItem /> renders properly with isGrid being false 1`] = `
<div>
  <div
    class="jsx-4127294762 carouselItem"
    data-testid="projectItem_touch_div_tjela logo"
    data-times-touched="0"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAwIiBoZWlnaHQ9IjQwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="tjela logo"
        decoding="async"
        src="/_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75"
        srcset="/_next/image?url=%2Fimages%2Ftjela.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Ftjela.jpg&w=828&q=75 2x"
        style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <div
      class="jsx-4127294762 backface"
    >
      <a
        class="jsx-4127294762 brand link"
        href="https://www.instagram.com/tudonatjela/"
        rel="noreferrer"
        target="_blank"
      >
        @tudonatjela
      </a>
      <p
        class="jsx-4127294762 description"
      >
        Gestão de Redes Sociais e Criação de Conteúdo
      </p>
      <p
        class="jsx-4127294762 date"
      >
        2020
         - Presente
      </p>
      <small
        class="jsx-4127294762"
      >
        * 
        em dupla com Sofia Ferreirinho
      </small>
    </div>
  </div>
</div>
`;
